calico/openstack/t_etcd.py

```diff
@@ -292,15 +292,21 @@ def profile_rules(self, profile_id):
                 # Put it all together and add to either the inbound or the
                 # outbound list.
                 if rule['direction'] == 'ingress':
-                    etcd_rule['src_tag'] = rule['remote_group_id']
-                    etcd_rule['src_net'] = net
-                    etcd_rule['src_ports'] = port_spec
+                    if rule['remote_group_id'] is not None:
+                        etcd_rule['src_tag'] = rule['remote_group_id']
+                    if net is not None:
+                        etcd_rule['src_net'] = net
+                    if port_spec is not None:
+                        etcd_rule['dst_ports'] = port_spec
```

Comment:
This line is where the actual fix is.  Ingress rules should match on dest port, not source port.